shap/plots/_image.py

```diff
@@ -127,3 +132,381 @@ def image(shap_values, pixel_values=None, labels=None, width=20, aspect=0.2, hsp
+
+    
+    # converting to json to be read in javascript
+    
+    image_data_json = json.dumps(shap_values.data.astype(int).tolist())
+    shap_values_color_dict_json = json.dumps(shap_values_color_dict)
+    image_data_gray_scale_json = json.dumps(image_data_gray_scale.astype(int).tolist())
+    
+    image_viz_html = f"""
+
+        <div id="{uuid}_image_viz" class="{uuid}_image_viz_content">
+          <div id="{uuid}_image_viz_header" style="padding:15px;margin:5px;font-family:sans-serif;font-weight:bold;">
+            <div style="display:inline">
+              <span style="font-size: 20px;"> Input/Output - Heatmap </span>
+            </div>
+          </div>
+          <div id="{uuid}_image_viz_content" style="display:flex;">
+            <div id="{uuid}_image_viz_input_container" style="padding:15px;border-style:solid;margin:5px;flex:2;">
+              <div id="{uuid}_image_viz_input_header" style="margin:5px;font-weight:bold;font-family:sans-serif;margin-bottom:10px">
+                Input Image
+              </div>
+              <div id="{uuid}_image_viz_input_content" style="margin:5px;font-family:sans-serif;">
+                  <canvas id="{uuid}_image_canvas" style="cursor:grab;width:100%;max-height:500px;"></canvas>
+                  <br>
+                  <br>
+                  <div id="{uuid}_tools">
+                      <div id="{uuid}_zoom">
+                        <span style="font-size:12px;margin-right:15px;"> Zoom </span>
+                        <button id="{uuid}_minus_button" class="zoom-button" onclick="{uuid}_zoom(-1)" style="background-color: #555555;color: white; border:none;font-size:15px;">-</button>
+                        <button id="{uuid}_plus_button" class="zoom-button" onclick="{uuid}_zoom(1)" style="background-color: #555555;color: white; border:none;font-size:15px;">+</button>
+                      </div>
+                      <br>
+                      <div id="{uuid}_opacity">
+                      <span style="font-size:12px;margin-right:15px;"> Image Opacity </span>
+                      <input type="range" min="1" max="100" value="35" style="width:100px" oninput="{uuid}_set_opacity(this.value)">
+                      </div>
+                  </div>
+              </div>
+            </div>
+            <div id="{uuid}_image_viz_output_container" style="padding:15px;border-style:solid;margin:5px;flex:1;">
+              <div id="{uuid}_image_viz_output_header" style="margin:5px;font-weight:bold;font-family:sans-serif;margin-bottom:10px">
+                Output Text
+              </div>
+              <div id="{uuid}_image_viz_output_content" style="margin:5px;font-family:sans-serif;">
+                  {output_text_html}
+              </div>
+            </div>
+          </div>
+        </div>
+
+    """
+    
+    
+    
+    image_viz_script = f"""
+        <script>
```

Comment:
I think there might be a better way to do this than embed the html/script code as strings, not sure how much effort we want to put into it though.  main problem is this is harder to read/understand as a string and there is no syntax highlighting.